ocaml/xapi/xapi_tunnel.ml

```diff
@@ -44,14 +44,17 @@ let create_internal ~__context ~transport_PIF ~network ~host ~protocol =
   let device_name = device in
   let mAC = Xapi_vif_helpers.gen_mac (0, Uuid.to_string (Uuid.make_uuid ())) in
   let metrics = Db.PIF.get_metrics ~__context ~self:transport_PIF in
+  let primary_address_type =
+    Db.PIF.get_primary_address_type ~__context ~self:transport_PIF
+  in
   Db.PIF.create ~__context ~ref:access_PIF
     ~uuid:(Uuid.to_string (Uuid.make_uuid ()))
     ~device ~device_name ~network ~host ~mAC ~mTU:(-1L) ~vLAN:(-1L) ~metrics
     ~physical:false ~currently_attached:false ~igmp_snooping_status:`unknown
     ~ip_configuration_mode:`None ~iP:"" ~netmask:"" ~gateway:"" ~dNS:""
     ~bond_slave_of:Ref.null ~vLAN_master_of:Ref.null ~management:false
     ~other_config:[] ~disallow_unplug:false ~ipv6_configuration_mode:`None
-    ~iPv6:[""] ~ipv6_gateway:"" ~primary_address_type:`IPv4 ~managed:true
+    ~iPv6:[""] ~ipv6_gateway:"" ~primary_address_type ~managed:true
```

Comment:
This looks good on the surface. What happens if the primary type is IPv6 when at the same time the IPv6 parameters are empty?

Reply:
I kept the same behavior as for `IPv4` (which fields are empty as well). It's up to the user to configure its PIFs as wanted once the tunnel is created.